### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by melashu**
# In this milestone we made the following changes 

- Add specs for the `factorial`, `reverse`, `fizzbuzz` method 
- Create a Solver class and implement the factorial`, `reverse`, `fizzbuzz` method

### CODE REVIEW

File: spec/tdd_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    it 'should return number' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(7)).to eq('7')
+    end
+  end
+  it 'Factorial method should return 120 when we pass 5' do
+    solver = Solver.new
+    expect(solver.factorial(5)).to eql 120
+  end
```
Comment (Omar-Muhamad) [inline spec/tdd_spec.rb:34]: To cover all test cases for the factorial method Please add the following tests that handle:
- [ ] Negative value which raises an exception as an expectation.
- [ ] Special case - when input is 0, the result should equal 1.

and then add those conditions to the factorial method in the solver class.

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @melashu and @shyusu4,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/tdd_spec.rb (hunk 1 above)
Comment (shyusu4) [reply spec/tdd_spec.rb]: Hi! Thank you for the review, we forgot to add those cases to our tests so we'll make sure to update our code and add them.

**Review by Omar-Muhamad — APPROVED** [review]
# Approved ✔
### Hi @melashu and @shyusu4,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**